src/condor_shadow.V6.1/NTreceivers.cpp

```diff
@@ -155,6 +155,10 @@ static const char * shadow_syscall_name(int condor_sysnum)
 	return "unknown";
 }
 
+// If we fail to send a reply to the starter, assume the socket is borked.
+// Close it and go into reconnect mode.
+#define ON_ERROR_RETURN(x) if (x == 0) {thisRemoteResource->disconnectClaimSock("Can no longer talk to condor_starter");return 0;}
```

Comment:
It won't break here, yet, but can we get an extra set of parens around the `x` in `x == 0`, just to be sure?